### ISSUE TITLE
Better type inference

### ISSUE THREAD

**Issue by EduardoVB** [twinbasic/twinbasic#1761]
**Is your feature request related to a problem? Please describe.**

In VB6, often developers find the issue of overflow, when working with literal values or operation with lower types.
For example, literals:

```
    Dim l As Long

    l = 300 * 300
```

Example of Long result from integers:

```
    Dim l As Long
    Dim i1 As Integer
    Dim i2 As Integer
    
    i1 = 300
    i2 = 300
    
    l = i1 * i1
```

This a a problem that often arises, leaving the developer clueless.

**Describe the solution you'd like**

Maybe tB could look for the type of the receiving variable?

**Additional context**

At vbforums, the question comes often, for example (some posts lately):
https://www.vbforums.com/showthread.php?902534
https://www.vbforums.com/showthread.php?878741&viewfull=1#post5630529

**Comment by fafalone** [comment]
I saw that thread too and was thinking about this... part of the problem here, is I bet this is a situation where some VB apps rely on the overflow to be triggered; e.g. to know when something is offscreen, maybe.

**Comment by fafalone** [comment]
Wouldn't be the strangest construct I've seen. 

Maybe it could be an option? Satisfy both needs. We turn it on, nobody wonders why their weird-technique using program from 30 years ago isn't working.

**Comment by EduardoVB** [comment]
I still would like to see a real world example of "using" that error in a program.

I think there are a lot more the cases of developers that don't understand why their programs are erroring and how to fix them.
It happened to me at one point, it most probably happened to you too, it happened to most of us. And it really makes little sense that a compiler could not handle that (from an intuitive POV).

**Comment by EduardoVB** [comment]
Maybe the option of erroring for integer overflows on Long type variable recipients should be the default option when importing a VB6 program, but not for new tB programs. What do you think?

**Comment by fafalone** [comment]
Yes but it should have a clear option; not be a hidden option where you can't change it unless you do or do not import a VBP.

**Comment by fafalone** [comment]
Need to do that for a few things IMO. It's currently set to turn DPI awareness on by default, that's going to result in improperly sized UIs for so many people. But I think Wayne has declared a vendetta against system DPI scaling 🤣
